File: app/src/main/java/org/tvheadend/tvhclient/data/service/worker/LoadChannelIconWorker.java
package org.tvheadend.tvhclient.data.service.worker;

import android.content.Context;
import android.content.Intent;
import androidx.annotation.NonNull;

import org.tvheadend.tvhclient.data.service.EpgSyncIntentService;

import androidx.work.Worker;
import androidx.work.WorkerParameters;
import timber.log.Timber;

public class LoadChannelIconWorker extends Worker {

    public LoadChannelIconWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {
        Timber.d("Loading channel icons from server");
        EpgSyncIntentService.enqueueWork(getApplicationContext(), new Intent().setAction("loadChannelIcons"));
        return Result.success();
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/util/worker/LoadChannelIconWorker.kt
package org.tvheadend.tvhclient.util.worker

import android.content.Context
import android.content.Intent
import androidx.work.Worker
import androidx.work.WorkerParameters
import org.tvheadend.tvhclient.service.HtspIntentService
import timber.log.Timber

class LoadChannelIconWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {

    companion object {
        const val WORK_NAME = "LoadChannelIconWorker"
    }

    override fun doWork(): Result {
        Timber.d("Loading channel icons from server")
        HtspIntentService.enqueueWork(applicationContext, Intent().setAction("loadChannelIcons"))
        return Result.success()
    }
}
